user-impl/src/main/java/com/example/auction/user/impl/PUserEntity.java

```diff
@@ -53,4 +54,21 @@ private Behavior notCreated() {
 
         return b.build();
     }
+    public static String hashPassword(String password_plaintext) {
+        String salt = BCrypt.gensalt(12);
+        String hashed_password = BCrypt.hashpw(password_plaintext, salt);
+
+        return (hashed_password);
+    }
+
+    public static boolean checkPassword(String password_plaintext, String stored_hash) {
+        boolean password_verified = false;
+
+        if (null == stored_hash)
+            throw new java.lang.IllegalArgumentException("Invalid hash provided for comparison");
+
+        password_verified = BCrypt.checkpw(password_plaintext, stored_hash);
```

Comment:
this could be replaced by 
```
return BCrypt.checkpw(password_plaintext, stored_hash);
```
and the `password_verified` var removed. 

(btw, the same in `hashPassword`)